src/frequenz/sdk/actor/power_distributing/_distribution_algorithm/_distribution_algorithm.py

```diff
@@ -84,21 +76,58 @@ def __init__(self, batteries: list[BatteryData]) -> None:
             self.soc_upper_bound = math.nan
             self.soc_lower_bound = math.nan
 
-        self.power_inclusion_upper_bound = sum(
-            b.power_inclusion_upper_bound for b in batteries
-        )
-        self.power_inclusion_lower_bound = sum(
-            b.power_inclusion_lower_bound for b in batteries
+        self.power_bounds = _aggregate_battery_power_bounds(
+            list(
+                map(
+                    lambda metrics: PowerBounds(
+                        inclusion_upper=metrics.power_inclusion_upper_bound,
+                        inclusion_lower=metrics.power_inclusion_lower_bound,
+                        exclusion_upper=metrics.power_exclusion_upper_bound,
+                        exclusion_lower=metrics.power_exclusion_lower_bound,
+                    ),
+                    batteries,
+                )
+            )
         )
-        # To satisfy the largest exclusion bounds in the set we need to
-        # provide the power defined by the largest bounds multiplied by the
-        # number of batteries in the set.
-        self.power_exclusion_upper_bound = max(
-            b.power_exclusion_upper_bound for b in batteries
-        ) * len(batteries)
-        self.power_exclusion_lower_bound = min(
-            b.power_exclusion_lower_bound for b in batteries
-        ) * len(batteries)
+
+
+def _aggregate_battery_power_bounds(
+    battery_metrics: Sequence[PowerBounds],
+) -> PowerBounds:
+    """Calculate bounds for a set of batteries located behind one set of inverters.
+
+    Args:
+        battery_metrics: List of PowerBounds for each battery.
+
+    Returns:
+        A PowerBounds object containing the aggregated bounds for all given batteries
+    """
+    assert len(battery_metrics) > 0, "No batteries given."
+
+    # Calculate the aggregated bounds for the set of batteries
+    power_inclusion_upper_bound = sum(
+        bounds.inclusion_upper for bounds in battery_metrics
+    )
+    power_inclusion_lower_bound = sum(
+        bounds.inclusion_lower for bounds in battery_metrics
+    )
+
+    # To satisfy the largest exclusion bounds in the set we need to
+    # provide the power defined by the largest bounds multiplied by the
+    # number of batteries in the set.
+    power_exclusion_upper_bound = max(
+        bounds.exclusion_upper for bounds in battery_metrics
+    ) * len(battery_metrics)
+    power_exclusion_lower_bound = min(
+        bounds.exclusion_lower for bounds in battery_metrics
+    ) * len(battery_metrics)
```

Comment:
This iterates 4 times through `battery_metrics`, this could probably impact on performance if there are many metrics. Please feel free to discard the command if the impact is not significant

Reply:
it's 4 metrics :)

Reply:
battery metrics here is usually not very much, usually it's `1` (the amount of batteries per battery-inverter set)